src/frontend/src/lib/components/send/SendInputAmount.svelte

```diff
@@ -0,0 +1,62 @@
+<script lang="ts">
+	import { Input } from '@dfinity/gix-components';
+	import { debounce, nonNullish } from '@dfinity/utils';
+	import { parseToken } from '$lib/utils/parse.utils';
+	import { slide } from 'svelte/transition';
+	import { i18n } from '$lib/stores/i18n.store';
+	import type { BigNumber } from '@ethersproject/bignumber';
+	import { invalidAmount } from '$lib/utils/input.utils';
+	import MaxButton from '$lib/components/common/MaxButton.svelte';
+
+	export let amount: number | undefined = undefined;
+	export let tokenDecimals: number | undefined = undefined;
+	export let placeholder: string = $i18n.core.text.amount;
+	export let customValidate: (userAmount: BigNumber) => Error | undefined = () => undefined;
+	export let calculateMax: (() => number | undefined) | null = null;
+
+	let error: Error | undefined;
+
+	let onMax = () => {
+		amount = calculateMax?.();
+	};
+
+	const validate = () => {
+		error = undefined;
+
+		if (invalidAmount(amount)) {
+			return;
+		}
+
+		const parsedValue = parseToken({
+			value: `${amount}`,
+			unitName: tokenDecimals
+		});
+
+		error = customValidate(parsedValue);
+	};
+
+	const debounceValidate = debounce(validate, 300);
+
+	$: amount, tokenDecimals, debounceValidate();
+</script>
+
+<label for="amount" class="font-bold px-4.5">{$i18n.core.text.amount}</label>
+<Input
+	name="amount"
+	inputType="currency"
+	required
+	bind:value={amount}
+	decimals={tokenDecimals}
+	{placeholder}
+	spellcheck={false}
+>
+	<svelte:component
```

Comment:
I would personally use a fragment here.

```
<svelte:fragment slot="inner-end">
    {#if nonNullish(calculateMax)}
        <MaxButton on:click={onMax} />
     {/if}
</svelte:fragment>
```

That way I would not create a `svelte:component` with `this=null`.

But it seems to work too, so just sharing the pattern in case you like it.

Reply:
agreed